### DISCUSSION TITLE
Source nuget declaration

### DISCUSSION THREAD

**Discussion by JPaja** [dotnet/csharplang#4400] [Language Ideas]
Main idea is to create way to share single file c# as script and share it as such, without need for `.csproj` for nuget links.

Possible solutions would be

```csharp

package Newtonsoft.Json;

using System;
using Newtonsoft.Json;

Console.WriteLine("Hello world!");
```

Possible keywords for nuget package
```
package
nuget
install
```
Another issue would be to specify version, where user could pick between concrete version like
```
12.0.3
13.0.1-beta1
```
or `latest`, `preview` (beta) and  `latestMajor`  (default)

Possible styles would be:

Comma separated
```csharp
package Newtonsoft.Json, 12.0.3;
package Newtonsoft.Json, 13.0.1-beta1;
package Newtonsoft.Json, latest;
package Newtonsoft.Json, preview;
```

Attribute
```csharp
[Version("latest")] 
package Newtonsoft.Json;

[Version("preview")] 
package Newtonsoft.Json;

[Version("12.0.3")]
package Newtonsoft.Json;

[Version("13.0.1-beta1")]
package Newtonsoft.Json;
```

**Comment by HaloFour** [comment]
IMO if there was a desire to add such support it could be done entirely with custom attributes and not require any specific language support.

```cs
[assembly: Package("Newtonsoft.Json", Version="latest")]
```

**Reply by canton7** [reply]
Things like dotnet-script and csx use e.g.

```cs
#r "nuget: AutoMapper, 6.1.0"
```

**Reply by JPaja** [reply]
Hmm yes dotnet-script seems nice also also I totally forgot about feature to set .net version and language version.
Maybe dotnet team can do some official  way to do single file scripts similar to dotnet-script
that would be officially supported by roslyn and dotnet

**Comment by CyrusNajmabadi** [comment]
FWIW when i see `package Newtonsoft.Json;` it totally makes me think you're *declaring* the package that your symbols will belong to.
